merge bundle/vat-admin devices, add waitForBundlecap()

warner opened the pull request:
fix(swingset): merge bundle/vat-admin devices, add waitForBundlecap()

Simplify the user experience by removing `devices.bundle` and merging its
functionality into vat-admin.

* bootstrap() can go back to doing `vatAdminService =
E(vats.vatAdmin).createVatAdminService(devices.vatAdmin)` ,
instead of `createVatAdminService(devices.vatAdmin, devices.bundle)`
* you can get bundlecaps from `E(vatAdminService).getBundlecap` or
`.getNamedBundlecap`, you no longer need `D(devices.bundle)` for those
* `E(vatAdminService).waitForBundlecap(bundleID)` gives you a Promise that
fires when (and if) the bundle has been installed, so you can e.g. begin a
`zoe.install()` before the bundle is installed (closes #4521)

closes #4566

erights reviewed (commented):
Why `*Bundlecap` instead of `*BundleCap`? Since none of this has been merged yet, if we're going to rename it, now is the ideal time.

warner commented:
> Why `*Bundlecap` instead of `*BundleCap`? Since none of this has been merged yet, if we're going to rename it, now is the ideal time.

Hrm, I guess it's an aesthetic thing, when I mentally pronounce the term, the capital `C` made me put stress on the "cap" syllable, and that felt intrusive. As if saying "bun-dle-CAP" is daring you to think that it being a capability is somehow unusual, and that the simplest === preferred form would be the shorter "bundle", and that's exactly what I'm trying to get away from here. Pronouncing it "BUN-dle-cap" seemed to emphasize the complete/indivisible-ness: it's a cap, it's just a bundle-specific cap. Not the strongest argument, I know.

To make `bundlecap` the most natural term, I changed the API and docs a little bit, to de-emphasize the places where we use all of `Bundle` and `Bundle[Cc]ap` and `BundleID` in the same place, to reduce the contrast and visibility of this issue.

The only place that `bundle` (a variable name) or `Bundle` (a type) should appear is in ZCF when it does
```js
const bundle = D(bundlecap).getBundle();
await importBundle(bundle);
```

and we hope to find a way to remove that eventually (to teach Endo about bundlecaps, and teach `xsnap` how to obtain their pre-parsed/compiled sources, with something like `await importBundle(bundlecap)`).

This leaves Zoe with some APIs and data structures that have both `bundlecap` (variable name) / `Bundle[Cc]ap` (type) and `bundleID` (variable name) / `BundleID` (type). "ID" reads better to me than `Id`, especially as a suffix, so I ruled out `bundleId`/`BundleId`. 

I dunno, I went back and forth on it once. I could revert the change and use `bundleCap`. Can you think of any consistency/aesthetic arguments for `bundleCap`?

erights commented:
> Can you think of any consistency/aesthetic arguments for bundleCap?

Just the obvious one: "bundlecap" is obviously the concatenation of two separate word-like identifiers "bundle" and "cap". We generally use bidcaps where these join. The exceptions are when the combined string has become so familiar as to be considered word-like on its own. I don't think "bundlecap" has that familiarity.

warner commented:
yeah, I think you're right.. I'll switch this to use `bundleCap`

FUDCo reviewed (approved):
![image](https://user-images.githubusercontent.com/8011352/155250554-c20ab1ec-421e-48ca-9693-65c423ebd1b6.png)

warner commented:
@dckc ok, so take a look at the second commit here. In `packages/vats/` you've got some type definitions that try to define what some swingset devices do. This PR changes the vatAdmin device into a "raw device", which lacks a `buildRootDeviceNode`. I added a typedef to describe what that object would have looked like (even though there is no actual object with that signature), and I changed `packages/vats/` to use it instead of trying to build something around the now-missing function.

We'll see if that passes `tsc` in CI. Please do a sanity check. And let's figure out a way to have swingset export this stuff properly, I suspect the approach in `packages/vats/` is fragile.

dckc commented:
> @dckc ... In `packages/vats/` you've got some type definitions that try to define what some swingset devices do. This PR changes the vatAdmin device ... Please do a sanity check.

Looks sane.

> And let's figure out a way to have swingset export this stuff properly, I suspect the approach in `packages/vats/` is fragile.

yes, let's. #4655
